src/Interpreters/Context.cpp

```diff
@@ -2250,6 +2276,28 @@ void Context::initializeTraceCollector()
     shared->initializeTraceCollector(getTraceLog());
 }
 
+#if USE_ROCKSDB
+void Context::initializeMergeTreeMetadataCache(const String & dir, size_t size)
+{
+    rocksdb::Options options;
+    rocksdb::BlockBasedTableOptions table_options;
+    rocksdb::DB * db;
+
+    options.create_if_missing = true;
+    options.statistics = rocksdb::CreateDBStatistics();
+    auto cache = rocksdb::NewLRUCache(size);
+    table_options.block_cache = cache;
+    options.table_factory.reset(rocksdb::NewBlockBasedTableFactory(table_options));
+    rocksdb::Status status = rocksdb::DB::Open(options, dir, &db);
+    if (status != rocksdb::Status::OK())
+    {
+        String message = "Fail to open rocksdb path at: " + dir + " status:" + status.ToString();
+        throw Exception(message, ErrorCodes::SYSTEM_ERROR);
```

Comment:
```suggestion
        throw Exception(ErrorCodes::SYSTEM_ERROR, "Fail to open rocksdb path at: {} status: {}", dir, status.ToString());
```

Reply:
Also maybe it worth removing/renaming this cache in case of corruption (under some setting maybe)